diff --git a/audio/drivers/alsathread.c b/audio/drivers/alsathread.c
index 030f23dad0..555ae515a1 100644
--- a/audio/drivers/alsathread.c
+++ b/audio/drivers/alsathread.c
@@ -140,7 +140,7 @@ static void *alsa_thread_init(const char *device,
    if (!alsa->info.fifo_lock || !alsa->info.cond_lock || !alsa->info.cond || !alsa->info.buffer)
       goto error;

-   alsa->info.worker_thread = sthread_create(alsa_worker_thread, alsa);
+   alsa->info.worker_thread = sthread_create_named_with_priority(alsa_worker_thread, alsa, "alsa", true);
    if (!alsa->info.worker_thread)
    {
       RARCH_ERR("[ALSA]: Failed to initialize worker thread\n");
diff --git a/gfx/video_thread_wrapper.c b/gfx/video_thread_wrapper.c
index ad369cd2e1..fdca796152 100644
--- a/gfx/video_thread_wrapper.c
+++ b/gfx/video_thread_wrapper.c
@@ -720,7 +720,7 @@ static bool video_thread_init(thread_video_t *thr,
    thr->suppress_screensaver = true;
    thr->last_time            = cpu_features_get_time_usec();

-   if (!(thr->thread = sthread_create(video_thread_loop, thr)))
+   if (!(thr->thread = sthread_create_named_with_priority(video_thread_loop, thr, "video", false)))
       return false;

    pkt.type                  = CMD_INIT;
diff --git a/libretro-common/rthreads/rthreads.c b/libretro-common/rthreads/rthreads.c
index ff11978668..87d9fb7c30 100644
--- a/libretro-common/rthreads/rthreads.c
+++ b/libretro-common/rthreads/rthreads.c
@@ -53,6 +53,7 @@
 #include "ctr_pthread.h"
 #else
 #include <pthread.h>
+#include <sched.h>
 #include <time.h>
 #endif

@@ -69,6 +70,8 @@
 #include <mach/mach.h>
 #endif

+#include "../../verbosity.h"
+
 struct thread_data
 {
    void (*func)(void*);
@@ -160,6 +163,30 @@ sthread_t *sthread_create(void (*thread_func)(void*), void *userdata)
 #define HAVE_THREAD_ATTR
 #endif

+sthread_t *sthread_create_named_with_priority(void (*thread_func)(void*), void *userdata, char *name, bool highprio)
+{
+   sthread_t *t = sthread_create(thread_func, userdata);
+   if (t) {
+      int policy = highprio ? SCHED_RR : SCHED_OTHER;
+      struct sched_param sparam = {
+         .sched_priority = sched_get_priority_max(policy),
+      };
+
+      int ret;
+      if ((ret = pthread_setname_np(t->id, name)) != 0) {
+         RARCH_ERR("sthread: pthread_setname_np error: %d\n", ret);
+      } else {
+         RARCH_LOG("sthread: set name '%s' for thread %u\n", name, t->id);
+      }
+      if ((ret = pthread_setschedparam(t->id, policy, &sparam)) != 0) {
+         RARCH_ERR("sthread: pthread_setschedparam error: %d\n", ret);
+      } else {
+         RARCH_LOG("sthread: set priority %d and policy '%s' for thread %u\n", sparam.sched_priority, (highprio ? "SCHED_RR" : "SCHED_OTHER"), t->id);
+      }
+   }
+   return t;
+}
+
 sthread_t *sthread_create_with_priority(void (*thread_func)(void*), void *userdata, int thread_priority)
 {
 #ifdef HAVE_THREAD_ATTR
