diff --git a/input/input_defines.h b/input/input_defines.h
index bdd36fc37a..60474b2fab 100644
--- a/input/input_defines.h
+++ b/input/input_defines.h
@@ -22,9 +22,9 @@

 #include <retro_common_api.h>
 
-#define MAX_USERS                      16
+#define MAX_USERS                      4
 
-#define MAX_INPUT_DEVICES              16
+#define MAX_INPUT_DEVICES              4
 
 #define RARCH_MAX_KEYS 137
 
