diff --git a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
index 90377193ed..2d501c56fe 100644
--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -2241,47 +2241,10 @@ static int frontend_unix_parse_drive_list(void *data, bool load_content)
       strlcpy(base_path + _len, "/.config/retroarch", sizeof(base_path) - _len);
    }
 #endif
-
-   {
-      size_t _len = strlcpy(udisks_media_path, "/run/media", sizeof(udisks_media_path));
-      if (user)
-      {
-         _len += strlcpy(udisks_media_path + _len, "/", sizeof(udisks_media_path) - _len);
-         strlcpy(udisks_media_path + _len, user, sizeof(udisks_media_path) - _len);
-      }
-   }
-
-   if (!string_is_empty(base_path))
-   {
-      menu_entries_append(list, base_path,
-            msg_hash_to_str(MENU_ENUM_LABEL_FILE_DETECT_CORE_LIST_PUSH_DIR),
-            enum_idx,
-            FILE_TYPE_DIRECTORY, 0, 0, NULL);
-   }
-   if (!string_is_empty(home))
-   {
-      menu_entries_append(list, home,
-            msg_hash_to_str(MENU_ENUM_LABEL_FILE_DETECT_CORE_LIST_PUSH_DIR),
-            enum_idx,
-            FILE_TYPE_DIRECTORY, 0, 0, NULL);
-   }
-   if (path_is_directory(udisks_media_path))
-   {
-      menu_entries_append(list, udisks_media_path,
-            msg_hash_to_str(MENU_ENUM_LABEL_FILE_DETECT_CORE_LIST_PUSH_DIR),
-            enum_idx,
-            FILE_TYPE_DIRECTORY, 0, 0, NULL);
-   }
-   if (path_is_directory("/media"))
-   {
-      menu_entries_append(list, "/media",
-            msg_hash_to_str(MENU_ENUM_LABEL_FILE_DETECT_CORE_LIST_PUSH_DIR),
-            enum_idx,
-            FILE_TYPE_DIRECTORY, 0, 0, NULL);
-   }
-   if (path_is_directory("/mnt"))
+   const char *roms_home = getenv("ROMS_HOME");
+   if (!string_is_empty(roms_home))
    {
-      menu_entries_append(list, "/mnt",
+      menu_entries_append(list, roms_home,
             msg_hash_to_str(MENU_ENUM_LABEL_FILE_DETECT_CORE_LIST_PUSH_DIR),
             enum_idx,
             FILE_TYPE_DIRECTORY, 0, 0, NULL);
