diff --git a/input/drivers/udev_input.c b/input/drivers/udev_input.c
index 865194e8b2..e15422234b 100644
--- a/input/drivers/udev_input.c
+++ b/input/drivers/udev_input.c
@@ -4025,7 +4025,7 @@ static void udev_input_free(void *data)
 }

 static bool open_devices(udev_input_t *udev,
-      enum udev_input_dev_type type, device_handle_cb cb)
+      enum udev_input_dev_type type, device_handle_cb cb, int limit)
 {
    struct udev_device *dev;
    const char             *type_str = g_dev_type_str[type];
@@ -4042,7 +4042,7 @@ static bool open_devices(udev_input_t *udev,
    udev_enumerate_scan_devices(enumerate);
    devs = udev_enumerate_get_list_entry(enumerate);

-   for (item = devs; item; item = udev_list_entry_get_next(item))
+   for (item = devs; item && limit > 0; item = udev_list_entry_get_next(item))
    {
       const char *devnode;
       const char *name = udev_list_entry_get_name(item);
@@ -4065,6 +4065,7 @@ static bool open_devices(udev_input_t *udev,
                      devnode, strerror(errno));

             close(fd);
+            limit--;
          }
       }
       udev_device_unref(dev);
@@ -4124,13 +4125,13 @@ static void *udev_input_init(const char *joypad_driver)

    udev->fd  = fd;

-   if (!open_devices(udev, UDEV_INPUT_KEYBOARD, udev_handle_keyboard))
+   if (!open_devices(udev, UDEV_INPUT_KEYBOARD, udev_handle_keyboard, 1))
       goto error;

-   if (!open_devices(udev, UDEV_INPUT_MOUSE, udev_handle_mouse))
+   if (!open_devices(udev, UDEV_INPUT_MOUSE, udev_handle_mouse, MAX_INPUT_DEVICES))
       goto error;

-   if (!open_devices(udev, UDEV_INPUT_TOUCHPAD, udev_handle_mouse))
+   if (!open_devices(udev, UDEV_INPUT_TOUCHPAD, udev_handle_mouse, MAX_INPUT_DEVICES))
       goto error;

 #ifdef UDEV_TOUCH_SUPPORT
