diff --git a/input/drivers/udev_input.c b/input/drivers/udev_input.c
index b16c4d3837..206ef76000 100644
--- a/input/drivers/udev_input.c
+++ b/input/drivers/udev_input.c
@@ -1332,7 +1332,7 @@ static void udev_input_free(void *data)
 }
 
 static bool open_devices(udev_input_t *udev,
-      enum udev_input_dev_type type, device_handle_cb cb)
+      enum udev_input_dev_type type, device_handle_cb cb, int limit)
 {
    const char             *type_str = g_dev_type_str[type];
    struct udev_list_entry     *devs = NULL;
@@ -1347,7 +1347,7 @@ static bool open_devices(udev_input_t *udev,
    udev_enumerate_scan_devices(enumerate);
    devs = udev_enumerate_get_list_entry(enumerate);
 
-   for (item = devs; item; item = udev_list_entry_get_next(item))
+   for (item = devs; item && limit > 0; item = udev_list_entry_get_next(item))
    {
       const char *name        = udev_list_entry_get_name(item);
 
@@ -1367,6 +1367,7 @@ static bool open_devices(udev_input_t *udev,
                      devnode, strerror(errno));
 
             close(fd);
+            limit--;
          }
       }
       udev_device_unref(dev);
@@ -1426,13 +1427,13 @@ static void *udev_input_init(const char *joypad_driver)
 
    udev->fd  = fd;
 
-   if (!open_devices(udev, UDEV_INPUT_KEYBOARD, udev_handle_keyboard))
+   if (!open_devices(udev, UDEV_INPUT_KEYBOARD, udev_handle_keyboard, 1))
       goto error;
 
-   if (!open_devices(udev, UDEV_INPUT_MOUSE, udev_handle_mouse))
+   if (!open_devices(udev, UDEV_INPUT_MOUSE, udev_handle_mouse, MAX_INPUT_DEVICES))
       goto error;
 
-   if (!open_devices(udev, UDEV_INPUT_TOUCHPAD, udev_handle_mouse))
+   if (!open_devices(udev, UDEV_INPUT_TOUCHPAD, udev_handle_mouse, MAX_INPUT_DEVICES))
       goto error;
 
    /* If using KMS and we forgot this,
